test/e2e/providers/nutanix/nutanix.go

```diff
@@ -0,0 +1,79 @@
+package nutanix
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	config "github.com/openshift/api/config/v1"
+	machinev1 "github.com/openshift/api/machine/v1"
+	machinev1beta1 "github.com/openshift/api/machine/v1beta1"
+	core "k8s.io/api/core/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	client "k8s.io/client-go/kubernetes"
+
+	"github.com/openshift/windows-machine-config-operator/test/e2e/clusterinfo"
+	"github.com/openshift/windows-machine-config-operator/test/e2e/providers/machineset"
+)
+
+type Provider struct {
+	oc *clusterinfo.OpenShift
+	*config.InfrastructureStatus
+}
+
+// New returns a new Provider
+func New(oc *clusterinfo.OpenShift, infraStatus *config.InfrastructureStatus) (*Provider, error) {
+	return &Provider{
+		oc:                   oc,
+		InfrastructureStatus: infraStatus,
+	}, nil
+}
+
+// GenerateMachineSet generates a Windows MachineSet which is Nutanix provider specific
+func (a *Provider) GenerateMachineSet(withIgnoreLabel bool, replicas int32) (*machinev1beta1.MachineSet, error) {
+	listOptions := meta.ListOptions{LabelSelector: "machine.openshift.io/cluster-api-machine-role=worker"}
+	machines, err := a.oc.Machine.Machines(clusterinfo.MachineAPINamespace).List(context.TODO(), listOptions)
+	if err != nil {
+		return nil, err
+	}
+	if len(machines.Items) == 0 {
+		return nil, fmt.Errorf("found 0 worker role machines")
+	}
+
+	workerSpec := &machinev1.NutanixMachineProviderConfig{}
+	err = json.Unmarshal(machines.Items[0].Spec.ProviderSpec.Value.Raw, workerSpec)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal raw machine provider spec: %v", err)
+	}
+
+	providerSpec := &machinev1.NutanixMachineProviderConfig{
+		Cluster:           workerSpec.Cluster,
+		Image:             workerSpec.Image,
```

Comment:
You need to reference a Windows image here. How are you doing that? [This](https://github.com/openshift/windows-machine-config-operator/blob/cc553e29d3bc16af106392e013e77c3ac4c5428d/test/e2e/providers/vsphere/vsphere.go#L68) is how we do it for vSphere. I suspect something similar will be needed for Nutanix.

Reply:
If the Nutanix cloud has Windows Server 2019 or 2022 images, you should reference that here. The `workerSpec.Image` will reference a Linux image.

Reply:
Yes, we need to pass a Windows image reference in L51 above. Let me check with our internal folks on which Windows image to use and fix this issue.